src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -103,11 +103,16 @@ def get_user_agent(self):
         """
 
         def parts():
-            if "user_agent" in self:
-                yield self.get("user_agent")
-            yield f"promptflow/{VERSION}"
+            agent = self.get("user_agent", '')
+            yield agent
+            promptflow_agent = f"promptflow/{VERSION}"
+            yield promptflow_agent if promptflow_agent not in agent else ''
+            user_agent = os.environ.get("USER_AGENT", '').strip()
+            yield user_agent if user_agent not in agent else ''
+            telemetry_agent = os.environ.get("TELEMETRY_AGENT", '').strip()
```

Comment:
what's TELEMETRY_AGENT?

Reply:
It is the newly added agent in the environment variable that we discussed yesterday
Of course, I'm not sure if I can directly use '**USER_AGENT**' in my business ', will this affect the previous business?

![image](https://github.com/microsoft/promptflow/assets/23182548/bd7851ba-70ab-4852-af20-98b5a55b7bb0)

Reply:
It is the newly added agent in the environment variable that we discussed yesterday
Of course, I'm not sure if I can directly use 'USER_ AGENT' in our business, will this affect the previous business?

![image](https://github.com/microsoft/promptflow/assets/23182548/0bdc0061-fd1e-46fc-a979-e96c2a4c985d)

Reply:
done